lib/src/widgets/utils/ds_group_card.widget.dart

```diff
@@ -0,0 +1,220 @@
+import 'package:blip_ds/blip_ds.dart';
+import 'package:blip_ds/src/widgets/chat/ds_message_bubble_detail.widget.dart';
+import 'package:flutter/material.dart';
+
+/// A Design System widget used to display a grouped [DSMessageBubble] list
+class DSGroupCard extends StatefulWidget {
+  final List<DSMessageItemModel> documents;
+  final Function compareMessages;
+  final bool isComposing;
+
+  /// Creates a new Design System's [DSGroupCard] widget
+  const DSGroupCard({
+    Key? key,
+    required this.documents,
+    required this.compareMessages,
+    required this.isComposing,
+  }) : super(key: key);
+
+  @override
+  State<DSGroupCard> createState() => _DSGroupCardState();
+}
+
+class _DSGroupCardState extends State<DSGroupCard> {
+  List<Widget> _widgets = [];
+
+  @override
+  Widget build(BuildContext context) {
+    _buildWidgetsList(_getGroupCards());
+
+    return Column(
+      children: _widgets,
+    );
+  }
+
+  void _buildWidgetsList(final List<Map<String, dynamic>> groups) {
+    _widgets = [];
+
+    for (var group in groups) {
+      int msgCount = 1;
+
+      group['msgs'].forEach(
+        (DSMessageItemModel message) {
+          int length = group['msgs'].length;
```

Comment:
Declare this variable as ```final```.

```suggestion
          final int length = group['msgs'].length;
```

Reply:
done